### ISSUE TITLE
Adding custom shading model(like cel-shading): Where should I start?

### ISSUE THREAD

**Issue by CodeLikeCXK** [GaijinEntertainment/DagorEngine#103]
For CPP part to shader then material

**Comment by AlexandrShcherbakov** [comment]
Hello!

Lighting is calculated in a frame graph node: https://github.com/GaijinEntertainment/DagorEngine/blob/main/prog/daNetGame/render/world/frameGraphNodes/gbufferNodes.cpp#L266
You can modify it to use your own shader for lighting.

Currently used shader is declared in https://github.com/GaijinEntertainment/DagorEngine/blob/main/prog/daNetGame/shaders/deferred_shadow_compute.dshl

**Comment by CodeLikeCXK** [comment]
I see. I would take a look. Thanks!

**Comment by CodeLikeCXK** [comment]
> Hello!
> 
> Lighting is calculated in a frame graph node: https://github.com/GaijinEntertainment/DagorEngine/blob/main/prog/daNetGame/render/world/frameGraphNodes/gbufferNodes.cpp#L266 You can modify it to use your own shader for lighting.
> 
> Currently used shader is declared in https://github.com/GaijinEntertainment/DagorEngine/blob/main/prog/daNetGame/shaders/deferred_shadow_compute.dshl

By the way...What is the relationship between daNetGame and rendering codes in samples like testGI?

**Comment by AlexandrShcherbakov** [comment]
These parts of code uses the same engine code and some common gameLibs. There are no direct connection between them.

**Comment by CodeLikeCXK** [comment]
> These parts of code uses the same engine code and some common gameLibs. There are no direct connection between them.

I see...
For example, if I wanted to make an RPG with stylized graphics. Which codebase should I base on, original engine code or daNetGame? Could I regard daNetGame as a subsidiary engine including not only game logic and system but also renderer?

**Comment by NicSavichev** [comment]
> > These parts of code uses the same engine code and some common gameLibs. There are no direct connection between them.
> 
> I see... For example, if I wanted to make an RPG with stylized graphics. Which codebase should I base on, original engine code or daNetGame? Could I regard daNetGame as a subsidiary engine including not only game logic and system but also renderer?

Yes, treat daNetGame as advanced framework that comes with customizable worldRenderer, ready set of shaders and multiplayer (client and dedicated) code.
Basis engine code (engine and gameLibs) will allow you to create your own framework but will take more efforts.
E.g. part of our games (WT, WTM) are based on basis engine (and in fact hold most of code inside game project itself) and part of our games are based on DNG (thus sharing large part of code so game projects contain shaders/render tweaks and gameplay specific code)
